src/main/kotlin/nl/connectplay/scoreplay/data/DatabaseGameRepository.kt

```diff
@@ -41,19 +44,130 @@ class DatabaseGameRepository(private val database: Database) : GameRepository {
+                    } else {
+                        stmt.setNull(8, java.sql.Types.DATE)
+                    }
+
+                    stmt.executeUpdate()
+
+                    // Read generated id
+                    val rs = stmt.generatedKeys
+                    val generatedId = if (rs.next()) rs.getInt(1) else throw IllegalStateException("Failed to retrieve generated id")
+                    rs.close()
+
+                    // Return created GameDto (without re-query; use provided fields + id)
+                    GameDto(
+                        id = generatedId,
+                        scoringMethodId = null,
+                        name = create.name,
+                        description = create.description,
+                        publisher = create.publisher,
+                        minPlayers = create.minPlayers,
+                        maxPlayers = create.maxPlayers,
+                        duration = create.duration,
+                        minAge = create.minAge,
+                        releaseDate = create.releaseDate
+                    )
+                }
+            } ?: throw IllegalStateException("No database connection")
+        }.await()
+    }
+
+override suspend fun updateGame(id: Int, update: UpdateGameDto): GameDto? = coroutineScope {
+    async {
+        database.connection?.use { conn ->
+            val sql = """
+                UPDATE games
+                SET
+                    name = COALESCE(?, name),
+                    description = COALESCE(?, description),
+                    publisher = COALESCE(?, publisher),
+                    minimum_player_count = COALESCE(?, minimum_player_count),
+                    maximum_player_count = COALESCE(?, maximum_player_count),
+                    duration = COALESCE(?, duration),
+                    minimum_age = COALESCE(?, minimum_age),
+                    release_date = COALESCE(?, release_date)
+                WHERE game_id = ?
+            """.trimIndent()
+
+            conn.prepareStatement(sql).use { stmt ->
+                if (update.name == null) stmt.setNull(2, Types.VARCHAR)
+                if (update.description == null) stmt.setNull(3, Types.VARCHAR)
+                if (update.publisher == null) stmt.setNull(4, Types.VARCHAR)
+                if (update.minPlayers == null) stmt.setNull(5, Types.INTEGER)
+                if (update.maxPlayers == null) stmt.setNull(6, Types.INTEGER)
+                if (update.duration == null) stmt.setNull(7, Types.INTEGER)
+                if (update.minAge == null) stmt.setNull(8, Types.INTEGER)
+
+                val sqlDate: java.sql.Date? = update.releaseDate?.let { d ->
+                    val jl = java.time.LocalDate.of(d.year, d.month, d.day)
+                    java.sql.Date.valueOf(jl)
+                }
+                if (sqlDate == null) stmt.setNull(8, Types.DATE)
+
+                // WHERE
+                stmt.setInt(1, id)
```

Comment:
het boeit JDBC niet of iets null is of niet - dat handelt de database af.

```suggestion
                stmt.setString(1, update.name)
                stmt.setString(2, update.description)
                stmt.setString(3, update.publisher)
                stmt.setInt(4, update.minPlayers)
                stmt.setInt(5, update.maxPlayers)
                stmt.setInt(6, update.duration)
                stmt.setInt(7, update.minAge)
                stmt.setDate(8,  update.releaseDate?.let {
                    Date.valueOf(it.toLocalDate())
                })

                // WHERE
                stmt.setInt(9, id)
```